libs/deal-ticket/src/components/deal-ticket/deal-ticket.tsx

```diff
@@ -84,8 +96,22 @@ export const DealTicket = ({ market, submit }: DealTicketProps) => {
     errors.summary?.type,
   ]);
 
+  useEffect(() => {
+    const priceUpdater = (event: CustomEvent) => {
```

Comment:
I'm not sure about this approach, I'd prefer to keep things in the react paradigm.

What I would suggest is to update the persisted order store to store a map of markets to orders, then we can just update the store as necessary and the deal ticket should receive the updates.

This also fits better with a suggestion I had for @asiaznik to update the local storage entries for the deal ticket to also be a map of markets to orders

Reply:
```
const [persistedOrder, setPersistedOrder] = usePersistedOrder(market);
const {
  register,
  ...
  setValue
} = useForm({
  defaultValues: persistedOrder || getDefaultOrder(market),
});

...

// store local form changes
useEffect(() => setPersistedOrder(order), [order, setPersistedOrder]);

// update form from external changes
useEffect(() => {
  // set changed fields
}, [persistedOrder])
```

Reply:
This does make me feel like react-hook-form doesn't provide a lot of value here, instead it means we have to maintain order state in two places

Reply:
So what you are suggesting was my initial approach and I discussed about this with @asiaznik as well. 

I ended up dispatching a custom event in order to avoid circular updating. 

The useEffect hooks from the example above would cause a circular update and the app will crash. 

The custom event seemed to be the simplest and cleanest solution.

Reply:
I see what you mean. Looks like the loop is caused by the persisted store and the useForm state relying on each other to stay in sync. I think it would be worth a refactor so that we have a single zustand store (with the persist middleware) for all orders across all markets. Then we should be able to just pass in the values to our already controlled form components. We may still need to update the form state after each update to ensure we get all the dirty field checking and validation that react-hook-form provides

Reply:
👋🏻 Folks, I have an idea on how to approach this with a little of the zustand magic.